### PR TITLE
completed grpc capability, BUUUUT..

### PR DESCRIPTION
**Pull request by bradygaster**
TL/DR: I don't think it is possible to host an HTTP 1 and an HTTP 2 endpoint on the same host. 

Adding @shirhatti here as he'd be able to confirm. At this point, this has 2 workers. 1, the original one, just watches the order queue and saves the order to the database. The 2nd worker basically runs the same time-based logic as from the original code, but as a worker. When any orders are less than a minute old, it sends out updates via gRPC. It all worked great, until I tried to put the gRPC endpoint into the web app - then, the worker pinging with the gRPC client failed miserably each time at accessing the endpoint. So there seems no way to get the message from the Worker to the web app via gRPC - seems like we'd need to add another app hosting the gRPC endpoint. 

I'm okay with that, but I have no idea how we'd get the message from the gRPC endpoint over to the HTTP 1 endpoint hosting the Blazor app. 

Open to ideas and input here. It is cool, *loved* the ease and simplicity of getting the protobuf models working over the wire.

### CODE REVIEW

**Comment by shirhatti** [comment]
> TL/DR: I don't think it is possible to host an HTTP 1 and an HTTP 2 endpoint on the same host.

Not true. In fact, by default our templates listen on both HTTP/1.1 and HTTP/2 by default.

I haven't yet looked at your PR, but I'm pretty sure the things that's tripping you up is that you tried to use gRPC without HTTPS.

**Comment by shirhatti** [comment]
To elaborate, the HTTP protocol version is selected by performing a handshake between the client and the server. This handshake is implemented as TLS extension, therefore you can only have both HTTP/1.1 and HTTP/2 enabled is use TLS.

If you don't use TLS, you're screwed. Since there is no handshake, the server doesn't know what to expect. This mode is called prior knowledge where basically the client and server agree upon a protocol beforehand.

**Comment by shirhatti** [comment]
You have 2 options in your case:
1. Use HTTPS with gRPC. Right now this is a god awful experience since you have to specify the public key of the cert used in the client. (The native gRPC client doesn't default to using trusted roots 😢 and ergo no PKI infrastructure. Just plain old fashioned mutual TLS)
  - https://github.com/grpc/grpc-dotnet/blob/master/examples/Shared/ClientResources.cs
  - https://github.com/grpc/grpc-dotnet/blob/master/examples/Clients/Greeter/Program.cs#L34
2. Create two endpoints one for HTTP/1 and one for HTTP/2 in the same app. It feels weird to me (granted I haven't reasoned about it much), but should more than get the job done. I'll create a sample that shows you how to do that (or send a PR. Haven't looked at your code yet).

**Comment by shirhatti** [comment]
> To elaborate, the HTTP protocol version is selected by performing a handshake between the client and the server. This handshake is implemented as TLS extension, therefore you can only have both HTTP/1.1 and HTTP/2 enabled is use TLS.

I also should've mentioned that this TLS extension (ALPN) isn't supported on OS X. Maybe you (@bradygaster) ran into that? (HTTP/2 server-side doesn't work on OS X)

**Comment by shirhatti** [comment]
Here's a sample https://github.com/shirhatti/Http1andHttp2/blob/master/Program.cs#L26-L35

Try this approach

File: src/BlazingPizza/BlazingPizza.csproj [hunk 1]
```diff
@@ -6,11 +6,18 @@
...
   </PropertyGroup>
 
+  <ItemGroup>
+    <Protobuf Include="Protos\pizzastatus.proto" GrpcServices="Server" Generator="MSBuild:Compile"/>
+    <Content Include="@(Protobuf)" />
+    <None Remove="@(Protobuf)" />
+  </ItemGroup>
+
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Authentication.MicrosoftAccount" Version="3.0.0-preview4-19216-03" />
```
Comment (shirhatti) [inline src/BlazingPizza/BlazingPizza.csproj]: preview4?

File: src/BlazingPizza/BlazingPizza.csproj [hunk 2]
```diff
@@ -19,4 +26,10 @@
     <ProjectReference Include="..\BlazingPizza.Shared\BlazingPizza.Shared.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Grpc.AspNetCore.Server" Version="0.1.20-pre1" />
+    <PackageReference Include="Google.Protobuf" Version="3.7.0" />
+    <PackageReference Include="Grpc.Tools" Version="1.20.0-pre3" PrivateAssets="All" />
```
Comment (shirhatti) [inline src/BlazingPizza/BlazingPizza.csproj]:
```suggestion
    <PackageReference Include="Grpc.Tools" Version="1.20.1" PrivateAssets="All" />
```
